Src/WitsmlExplorer.Api/Services/CredentialsService.cs

```diff
@@ -145,9 +146,10 @@ public bool ValidEncryptedBasicCredentials(string headerName)
         public async Task<(string username, string witsmlUsername)> GetUsernames()
         {
             StringValues authorizationHeader = _httpContextAccessor.HttpContext.Request.Headers["Authorization"];
-            string bearerToken = authorizationHeader.Count > 0 ? authorizationHeader.ToString().Split()[1] : null;
+            string bearerToken = authorizationHeader.Count > 0 ? authorizationHeader.ToString().Split()[1] : string.Empty;
+            bearerToken = Regex.Match(bearerToken, "[a-zA-Z0-9-_.]+").Value;
             ServerCredentials creds = await GetCredentials(WitsmlClientProvider.WitsmlTargetServerHeader, bearerToken);
-            if (bearerToken != null)
+            if (string.IsNullOrEmpty(bearerToken))
```

Comment:
Logic was inverted from "bearerToken is not null" to "bearerToken is null" - missing exclamation mark